sdk/python/feast/go_server.py

```diff
@@ -249,96 +216,71 @@ def get_online_features(
+        self._go_server_background_thread.stop()
 
 
-# https://www.geeksforgeeks.org/python-different-ways-to-kill-a-thread/
-class GoServerBackgroundThread(threading.Thread):
+class GoServerMonitorThread(threading.Thread):
     def __init__(
         self,
         name: str,
         shared_connection: GoServerConnection,
-        go_server_started: threading.Event,
+        go_server_first_started: threading.Event,
     ):
         threading.Thread.__init__(self)
         self.name = name
         self._shared_connection = shared_connection
-        self._go_server_started = go_server_started
+        self._is_cancelled = False
+        self.daemon = True
+        self._go_server_started = go_server_first_started
 
     def run(self):
         # Target function of the thread class
+        _logger.info("Started GoServerMonitorThread")
         try:
-            while True:
-                self._go_server_started.clear()
+            while not self._is_cancelled:
 
                 # If we fail to connect to grpc stub, terminate subprocess and repeat
                 if not self._shared_connection.connect():
                     self._shared_connection.kill_process()
                     continue
-                self._go_server_started.set()
-                while True:
+                else:
+                    _logger.info("Set go server first started event")
+                    self._go_server_started.set()
+                while not self._is_cancelled:
                     try:
                         # Making a blocking wait by setting timeout to a very long time so we don't waste cpu cycle
                         self._shared_connection.wait_for_process(3600)
                     except subprocess.TimeoutExpired:
                         pass
+
+                    _logger.info(
+                        "Process state: %s %s",
+                        self._shared_connection._process.pid,
+                        self._shared_connection._process.returncode,
+                    )
                     if not self._shared_connection.is_process_alive():
                         break
         finally:
             # Main thread exits
             self._shared_connection.kill_process()
 
-    def stop_go_server(self):
-        thread_id = self._get_id()
-        res = ctypes.pythonapi.PyThreadState_SetAsyncExc(
-            thread_id, ctypes.py_object(SystemExit)
-        )
-        # TODO: Review that kill process here but run also has to stop
-        if res > 1:
-            ctypes.pythonapi.PyThreadState_SetAsyncExc(thread_id, 0)
-        else:
-            self._shared_connection.kill_process()
-
-    def _get_id(self):
-        # returns id of the respective thread
-        if hasattr(self, "_thread_id"):
-            return self._thread_id
-        for id, thread in threading._active.items():
-            if thread is self:
-                return id
+    def stop(self):
+        _logger.info("Stopping GoServerMonitorThread")
```

Comment:
Is it clearer to use plain english, do our users know what `GoServerMonitorThread` is?

Reply:
updated the log, but made it debug logging too.